src/util/map.test.js

```diff
@@ -294,4 +294,35 @@ describe("util/map", () => {
       ]);
     });
   });
+  describe("pushValue", () => {
+    it("works when the map has no matching key", () => {
+      const map = new Map();
+      MapUtil.pushValue(map, "foo", 3);
+      expect(map.get("foo")).toEqual([3]);
+    });
+    it("works when the map has a matching key", () => {
+      const map = new Map();
+      MapUtil.pushValue(map, "foo", 3);
+      MapUtil.pushValue(map, "foo", 4);
+      expect(map.get("foo")).toEqual([3, 4]);
+    });
+    it("works when the map already has an empty array", () => {
+      const map = new Map().set("foo", []);
+      MapUtil.pushValue(map, "foo", 1);
+      expect(map.get("foo")).toEqual([1]);
+    });
+    it("preserves array identity", () => {
+      const arr = [];
+      const map = new Map().set("foo", arr);
+      MapUtil.pushValue(map, "foo", 1);
+      expect(map.get("foo")).toBe(arr);
+    });
+    it("doesn't add anything else to the map", () => {
+      const map = new Map();
+      MapUtil.pushValue(map, "foo", 3);
+      MapUtil.pushValue(map, "foo", 4);
+      expect(Array.from(map.entries())).toEqual([["foo", [3, 4]]]);
```

Comment:
Are these two `expect`s a vestige of a `class`ier implementation? The
first one seems okay, though it’d be better\* to write

```js
expect(map).toEqual(new Map().set("foo", [3, 4]));
```

But the second `expect` is quite confusing and should probably be
removed, unless I’m missing something.

\* “better” because this does not depend on the order of the key–value
pairs returned by `entries`—which is trivial in this case of a size-`1`
map, thus merely “better” and not “mandatory”. :-)

Reply:
Changed.